source/lib/vagrant-openstack-provider/config.rb

```diff
@@ -371,6 +398,17 @@ def validate(machine)
 
       private
 
+      def validate_api_version(errors)
+        if @identity_api_version == '2'
+          errors << I18n.t('vagrant_openstack.config.tenant_name_required') if @tenant_name.nil? || @tenant_name.empty?
+          errors << I18n.t('vagrant_openstack.config.invalid_endpoint_type') unless  %w(publicURL adminURL internalURL).include?(@endpoint_type)
+        elsif @identity_api_version == '3'
+          errors << I18n.t('vagrant_openstack.config.domain_required') if @domain_name.nil? || @domain_name.empty?
+          errors << I18n.t('vagrant_openstack.config.project_name_required') if @project_name.nil? || @project_name.empty?
+          errors << I18n.t('vagrant_openstack.config.invalid_interface_type') unless  %w(public admin internal).include?(@interface_type)
```

Comment:
Those 3 strings must have a textual message in `https://github.com/ggiamarchi/vagrant-openstack-provider/blob/master/source/locales/en.yml`